crates/polars-core/src/serde/series.rs

```diff
@@ -225,6 +255,17 @@ impl<'de> Deserialize<'de> for Series {
                             .cast(&DataType::Categorical(None))
                             .unwrap())
                     },
+                    #[cfg(feature = "dtype-array")]
+                    DataType::Array(dt, size) => unsafe {
+                        let mut values: Vec<Series> = map.next_value()?;
+                        let mut builder =
+                            get_fixed_size_list_builder(&dt, values.len(), size, &name).unwrap();
+                        for (i, s) in values.iter_mut().enumerate() {
+                            let s = s.rechunk();
+                            builder.push_unchecked(s.to_arrow(0).as_ref(), i)
```

Comment:
Given that we are deserializing here, we also nee the `safe` variants that check the invariants on the builder.

I have only written the unsafe variants now for internal use.

The safe variants can do an invariant check and dispatch to the unsafe ones.